internals/src/Tizen.Applications.ThemeManager/Interop/Interop.ThemeManager.cs

```diff
@@ -0,0 +1,133 @@
+            OutOfMemory = Tizen.Internals.Errors.ErrorCode.OutOfMemory,
+            PermissionDenied = Tizen.Internals.Errors.ErrorCode.PermissionDenied,
+            IoError = Tizen.Internals.Errors.ErrorCode.IoError,
+        }
+
+        internal static class ThemeManagerErrorFactory
+        {
+            internal static Exception GetException(Interop.ThemeManager.ErrorCode err, string message)
+            {
+                string errMessage = string.Format("{0} err = {1}", message, err);
+                Log.Warn(LogTag, errMessage);
+                switch (err)
+                {
+                    case Interop.ThemeManager.ErrorCode.InvalidParameter:
+                        return new ArgumentException(errMessage);
+                    case Interop.ThemeManager.ErrorCode.OutOfMemory:
+                        return new OutOfMemoryException(errMessage);
+                    case Interop.ThemeManager.ErrorCode.PermissionDenied:
+                        return new UnauthorizedAccessException(errMessage);
+                    case Interop.ThemeManager.ErrorCode.IoError:
+                    default:
+                        return new InvalidOperationException(errMessage);
+                }
+            }
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        internal delegate void ThemeLoaderChangedCallback(IntPtr handle, IntPtr userData);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_id")]
+        internal static extern ErrorCode GetId(IntPtr handle, out string id);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_version")]
+        internal static extern ErrorCode GetVersion(IntPtr handle, out string version);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_tool_version")]
+        internal static extern ErrorCode GetToolVersion(IntPtr handle, out string version);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_title")]
+        internal static extern ErrorCode GetTitle(IntPtr handle, out string title);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_resolution")]
+        internal static extern ErrorCode GetResolution(IntPtr handle, out string resolution);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_preview")]
+        internal static extern ErrorCode GetPreview(IntPtr handle, out string preview);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_description")]
+        internal static extern ErrorCode GetDescription(IntPtr handle, out string description);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_string")]
+        internal static extern ErrorCode GetString(IntPtr handle, string key, out string val);
+
+        [DllImport(Libraries.ThemeManager, EntryPoint = "theme_get_string_array")]
+        internal static extern ErrorCode GetStringArray(IntPtr handle, string key, out string[] val);
```

Comment:
I wonder is it working well, 
We usually use IntPtr for out param of string array, 
like `Interop.AppControl.GetExtraDataArray`
``` c#
        [DllImport(Libraries.AppControl, EntryPoint = "app_control_get_extra_data_array")]
        internal static extern ErrorCode GetExtraDataArray(SafeAppControlHandle handle, string key, out IntPtr value, out int length);
```

Reply:
Done